src/clj/fluree/db/datatype.cljc

```diff
@@ -97,6 +106,12 @@
      (number? x)  const/$xsd:decimal
      (boolean? x) const/$xsd:boolean)))
 
+(defn infer-iri
+  ([x]
+   (some-> x infer iri/sid->iri))
+  ([x lang]
```

Comment:
It doesn't look like this arity is ever used, do we need it?

Reply:
Hmmm. I think we do need it, but you're right that it currently isn't used. I think it _should_ be used in some places though. I'll have to check why it isn't. Thanks for pointing this out.

Reply:
I think I was thrown off by the name of the function, I initially thought this was inferring a datatype of xsd:anyURI. Maybe it would be more clear if it was called `infer-dt-iri`?

Reply:
that makes sense.

Reply:
I actually think this would be redundant on second thought. This namespace is `fluree.db.datatype`, and would generally be referred as `datatype`, so any call site of this function would be `datatype/infer-iri`.

I think the information of what this function does is well encoded in both its name and the namespace it's in, but let me know if you feel strongly otherwise and I'll change it.